### ISSUE TITLE
Unknown block type 

### ISSUE THREAD

**Issue by benjamindamm** [google/blockly#5867]
After using Version **7.20211209.2**

```
core.mjs:6469 ERROR TypeError: Unknown block type: controls_if
    at BlockSvg.Block (:4000/vendor.js:17284:98)
    at new BlockSvg (:4000/vendor.js:17383:191)
    at WorkspaceSvg.module$exports$Blockly$WorkspaceSvg.WorkspaceSvg.newBlock (:4000/vendor.js:17510:90)
    at module$contents$Blockly$serialization$blocks_appendPrivate (:4000/vendor.js:16692:291)
    at module$exports$Blockly$serialization$blocks.appendInternal (:4000/vendor.js:16691:7)
    at module$exports$Blockly$serialization$blocks.append (:4000/vendor.js:16689:154)
    at VerticalFlyout.module$exports$Blockly$Flyout.Flyout.createFlyoutBlock_ (:4000/vendor.js:17604:477)
    at VerticalFlyout.module$exports$Blockly$Flyout.Flyout.createFlyoutInfo_ (:4000/vendor.js:17600:423)
    at VerticalFlyout.module$exports$Blockly$Flyout.Flyout.show (:4000/vendor.js:17598:300)
    at Toolbox.module$exports$Blockly$Toolbox.Toolbox.updateFlyout_ (:4000/vendor.js:17702:146)
```

This error comes up. Seems that I cant use my toolbox definition as I am used to. In this section of the code the blocks seem to be undefined.

```
if (b) {
        this.type = b;
        c = $.module$exports$Blockly$blocks.Blocks[b];
        if (!c || 'object' !== typeof c)
          throw TypeError('Unknown block type: ' + b);
        (0, $.module$exports$Blockly$utils$object.mixin)(this, c);
      }
```

This is how I define those blocks:

```
export const defaultToolbox: Blockly.utils.toolbox.ToolboxInfo = {
  kind: 'categoryToolbox',
  contents: [
    // Category: Logik
    {
      id: null,
      categorystyle: null,
      colour: '#536f92',
      cssconfig: null,
      hidden: false,
      kind: 'category',
      name: 'EDITOR.LOGIC',
      contents: [
        {
          kind: 'block',
          type: 'controls_if'
        },
        {
          kind: 'block',
          type: 'controls_ifelse'
        },
        {
          kind: 'block',
          type: 'logic_compare'
        },
        {
          kind: 'block',
          type: 'logic_operation'
        },
        {
          kind: 'block',
          type: 'logic_boolean'
        },
        {
          kind: 'block',
          type: 'logic_negate'
        },
      ],
    },
    // Category: Values
    {
      id: null,
      categorystyle: null,
      colour: '#536f92',
      cssconfig: null,
      hidden: false,
      kind: 'category',
      name: 'EDITOR.VALUES',
      contents: [
        {
          kind: 'block',
          type: 'math_number'
        },
        {
          kind: 'block',
          type: 'text'
        },
        {
          kind: 'block',
          type: 'text_length'
        },
        {
          kind: 'block',
          type: 'text_isEmpty'
        },
      ],
    },
    // Category: Variables
    {
      id: null,
      categorystyle: null,
      colour: '#536f92',
      cssconfig: null,
      hidden: false,
      kind: 'category',
      name: 'EDITOR.VARIABLES',
      contents: [],
      custom: 'VARIABLE',
    },
  ],
};
```

The main problem is, that there are breaking changes. But I can't find any documentation to understand what needs to be changed.

**Comment by gonfunko** [comment]
Does a change similar to https://github.com/google/blockly/pull/5797/files (adding a call to goog.require('Blockly.blocks.all') after you load Blockly and before you configure your toolbox) help?

**Comment by mishranshi2309** [comment]
I am facing the same issue i.e "Unknown block types" for my custom blocks

**Comment by alschmiedt** [comment]
Can you please provide some more information on how you are importing the blocks? Are you using the script tag `blocks_compressed.js`? Or are you using `import 'blockly/blocks`? And can you provide what version of Blockly you are using?

**Comment by mishranshi2309** [comment]
> 

I am using blockly 6.20210701.0 . 
I am integrating it with angular 12

I have a xml string to render toolbar 
![Capture](https://user-images.githubusercontent.com/73482316/154623764-7b558cde-43db-430d-b5cc-3bf28b8d5aaa.PNG)

the component.ts file looks like:
![ts](https://user-images.githubusercontent.com/73482316/154624117-7d00acee-d37d-432e-a524-d606a5d7c75b.PNG)

the custom block is in different file in asset folder 
![selector](https://user-images.githubusercontent.com/73482316/154624234-c5edae7c-c37d-4494-9556-14bd8bfd70f4.PNG)

and had included the path in angular.json
 "scripts": [
              "node_modules/blockly/blockly_compressed.js",
               "node_modules/blockly/python_compressed.js",
               "node_modules/blockly/javascript_compressed.js",
               "node_modules/blockly/php_compressed.js",
               "src/assets/lib/sql.js",
               "src/assets/lib/maps.js"
            ]

**Comment by alschmiedt** [comment]
Hello! I just want to preface this with a disclaimer that I am not in any way an expert in angular, and you might have better luck asking the question on the [forum](https://groups.google.com/g/blockly). Another developer there with more angular experience might be able to help.

That being said,  I looked into this a bit and tried to reproduce using the [blockly angular example](https://github.com/google/blockly-samples/tree/master/examples/blockly-angular). I was able to get it working by adding `import '../../assets/sql';` to the equivalent of your `component.ts` file. And adding `import * as Blockly from 'blockly';
` to the top of `sql.js`. 

I believe the problem with putting it in the "scripts" path is that they are "loaded exactly as if you had added them in a <script> tag inside index.html". Meaning that I believe you are including two different instances of Blockly. The one you are importing is different than the one used inside `sql.js`, so when you go to look for your custom block it is not there because it was never added.
